### PR TITLE
Uniform logging for sdk automation

### PR DESCRIPTION
**Pull request by raych1**
- uniform error messages by adding consistent prefix and next steps
- filter logging in sdk automation framework

Test PRs:
- one service
  https://github.com/test-repo-billy/azure-rest-api-specs/pull/3422/checks?check_run_id=27889478895
- multiple services
  https://github.com/test-repo-billy/azure-rest-api-specs/pull/3421/checks?check_run_id=27889362992

### CODE REVIEW

File: eng/scripts/automation/GenerateAndBuildLib.ps1 [hunk 1]
```diff
@@ -750,18 +762,22 @@ function GeneratePackage()
...
 
-    if ( $?) {
-        Write-Host "Start to generate sdk package $projectFolder"
+    if ($isGenerateSuccess) {
         # Build project when successfully generated the code
         Write-Host "Start to build sdk project: $srcPath"
         dotnet build $srcPath /p:RunApiCompat=$false
         if ( !$?) {
-            Write-Error "Failed to build project. exit code: $?"
+            Write-Host "[ERROR] Failed to build the sdk project for service: $service with exit code: $?."
```
Comment (chunyu3) [inline eng/scripts/automation/GenerateAndBuildLib.ps1:778]:
```suggestion
            Write-Host "[ERROR] Failed to build the sdk project: $packageName for service: $service with exit code: $?."
```

File: eng/scripts/automation/GenerateAndBuildLib.ps1 [hunk 2]
```diff
@@ -770,29 +786,44 @@ function GeneratePackage()
...
+                Write-Host "[ERROR] If the issue persists, contact the DotNet language support channel at $DotNetSupportChannelLink and include this spec pull request."
                 $result = "failed"
             }
             # pack
             Write-Host "Start to pack sdk"
             dotnet pack $srcPath /p:RunApiCompat=$false
             if ( !$? ) {
-                Write-Error "Failed to pack sdk. exit code: $?"
+                Write-Host "[ERROR] Failed to pack the following sdk with exit code: $?."
+                Write-Host "[ERROR] $srcPath."
```
Comment (chunyu3) [inline eng/scripts/automation/GenerateAndBuildLib.ps1:799]: We should pass `$srcPath` to up line. It is strange that list a path alone in a line

File: eng/scripts/automation/GenerateAndBuildLib.ps1 (hunk 1 above)
Comment (raych1) [reply eng/scripts/automation/GenerateAndBuildLib.ps1]: Fixed.

File: eng/scripts/automation/GenerateAndBuildLib.ps1 (hunk 2 above)
Comment (raych1) [reply eng/scripts/automation/GenerateAndBuildLib.ps1]: Added packageName and service to the output.

**Review by chunyu3 — APPROVED** [review]

File: eng/swagger_to_sdk_config.json [hunk 3]
```diff
@@ -2,7 +2,15 @@
   "$schema": "https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/documentation/sdkautomation/SwaggerToSdkConfigSchema.json",
   "generateOptions": {
     "generateScript": {
-      "path": "pwsh ./eng/scripts/Invoke-GenerateAndBuildV2.ps1"
+      "path": "pwsh ./eng/scripts/Invoke-GenerateAndBuildV2.ps1",
+      "logPrefix": "DotNet-Sdk-Automation",
+      "stderr": {
+        "showInComment": ".*(\\[ERROR\\]|error NU|(?!0\\b)\\d+ Error\\(s\\)|ExternalError).*",
+        "scriptError": "\\[ERROR\\]"
```
Comment (weshaggard) [inline eng/swagger_to_sdk_config.json:9]: Do we need to also include errors that come from msbuild which generally look like "error: "?

**Pull request merged**